nvflare/app_opt/tracking/mlflow/mlflow_receiver.py

```diff
@@ -0,0 +1,324 @@
+
+        art_full_path = self.get_artifact_location(self.artifact_location)
+        experiment_name = self.kwargs.get(TrackConst.EXPERIMENT_NAME, "FLARE FL Experiment")
+        experiment_tags = self._get_tags(TrackConst.EXPERIMENT_TAGS, kwargs=self.kwargs)
+
+        sites = fl_ctx.get_engine().get_clients()
+        self._init_buffer(sites)
+        self.mlflow_setup(art_full_path, experiment_name, experiment_tags, sites)
+
+    def mlflow_setup(self, art_full_path, experiment_name, experiment_tags, sites):
+        """Set up an MlflowClient for each client site and create an experiment and run.
+
+        Args:
+            art_full_path (str): Full path to artifacts.
+            experiment_name (str): Experiment name.
+            experiment_tags (dict): Experiment tags.
+            sites (List[Client]): List of client sites.
+        """
+        for site in sites:
+            mlflow_client = self.mlflow_clients.get(site.name, None)
+            if not mlflow_client:
+                mlflow_client = MlflowClient()
+                self.mlflow_clients[site.name] = mlflow_client
+                self.experiment_id = self._create_experiment(
+                    mlflow_client, experiment_name, art_full_path, experiment_tags
+                )
+                run_group_id = str(int(time.time()))
+
+                default_run_name = "FLARE FL Run"
+                run_name = self.get_run_name(self.kwargs, default_run_name, site.name, run_group_id)
+                tags = self.get_run_tags(self.kwargs, run_group_id, run_name)
+                run = mlflow_client.create_run(experiment_id=self.experiment_id, run_name=run_name, tags=tags)
+                self.run_ids[site.name] = run.info.run_id
+
+    def _init_buffer(self, sites):
+        """For each site, create a buffer (dict) consisting of a list each for metrics, parameters, and tags."""
+        for site in sites:
+            self.buffer[site.name] = {
+                AnalyticsDataType.METRICS: [],
+                AnalyticsDataType.PARAMETERS: [],
+                AnalyticsDataType.TAGS: [],
+            }
+
+    def get_run_name(self, kwargs: dict, default_name: str, site_name: str, run_group_id: str):
+        run_name = kwargs.get(TrackConst.RUN_NAME, default_name)
+        job_id_tag = self.get_job_id_tag(group_id=run_group_id)
+        return f"{site_name}-{job_id_tag}-{run_name}"
+
+    def get_run_tags(self, kwargs, run_group_id, run_name: str):
+        run_tags = self._get_tags(TrackConst.RUN_TAGS, kwargs=kwargs)
+        run_tags["job_id"] = self.get_job_id_tag(group_id=run_group_id)
+        run_tags["run_name"] = run_name
+        return run_tags
+
+    def get_job_id_tag(self, group_id: str) -> str:
+        job_id = self.fl_ctx.get_job_id()
+        if job_id == "simulate_job":
+            # Since all jobs run in the simulator have the same job_id of "simulate_job", use group_id instead
+            job_id = group_id
+        return job_id
```

Comment:
Should we do this part here?

If the all jobs run in the simulator have the same job_id of "simulate_job", then that is simulator's issue/implementation.

We should just use job_id here IMO.

If we want simulator to use new job name or custom job name, we should modify simulator?

Reply:
This is to make the UI of mlflow have different run names, which is pretty important if there are multiple separate runs. I think in principle what you say makes sense but for now maybe keep this as is. I may further refine how this is as well later.